integrationTests/chainSimulator/bridge/bridgeAndTransfer_test.go

```diff
@@ -79,57 +81,144 @@ func TestChainSimulator_ExecuteMintBurnBridgeOpForESDTTokensWithPrefixAndTransfe
-	chainSim.RequireAccountHasToken(t, cs, tokenToBridge, esdtSafeEncoded, big.NewInt(0))
+	for _, bridgedInToken := range groupTokens(bridgedInTokens) {
+		checkTokenDataInAccount(t, cs, bridgedInToken, esdtSafeEncoded, big.NewInt(0))       // sender
+		checkTokenDataInAccount(t, cs, bridgedInToken, wallet.Bech32, bridgedInToken.Amount) // receiver
+	}
 
-	tokenSupply, err := nodeHandler.GetFacadeHandler().GetTokenSupply(tokenToBridge)
+	// Generate address in another shard than bridge contract and user wallet
+	receiver, err := cs.GenerateAndMintWalletAddress(2, chainSim.InitialAmount)
 	require.Nil(t, err)
-	require.NotNil(t, tokenSupply)
-	require.Equal(t, amountToDeposit.String(), tokenSupply.Burned)
-
-	// Send some of the bridged prefixed tokens to another address
-	receiver := "erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx"
-	receiverBytes, err := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode(receiver)
+	receiverNonce := uint64(0)
+	receiver2, err := cs.GenerateAndMintWalletAddress(2, chainSim.InitialAmount)
 	require.Nil(t, err)
 
-	receivedTokens := big.NewInt(11)
-	chainSim.TransferESDT(t, cs, wallet.Bytes, receiverBytes, &nonce, tokenToBridge, receivedTokens)
-	chainSim.RequireAccountHasToken(t, cs, tokenToBridge, receiver, receivedTokens)
-	chainSim.RequireAccountHasToken(t, cs, tokenToBridge, wallet.Bech32, big.NewInt(0).Sub(remainingValueAfterBridge, receivedTokens))
+	// will make a cross shard transaction, then deposit transaction, then same shard transaction with every token
+	for _, token := range groupTokens(bridgedInTokens) {
+		amountToSend := big.NewInt(2)
+		if token.Type == core.NonFungibleV2 || token.Type == core.DynamicNFT {
+			amountToSend = big.NewInt(1)
+		}
+		// cross shard transfer
+		receiverToken := transferToken(t, cs, token, wallet, receiver, &nonce, amountToSend)
+		checkTokenDataInAccount(t, cs, token, wallet.Bech32, big.NewInt(0).Sub(token.Amount, receiverToken.Amount))
+		checkTokenDataInAccount(t, cs, token, receiver.Bech32, receiverToken.Amount)
+
+		amountToDeposit := big.NewInt(1)
+		bridgedOutToken := chainSim.ArgsDepositToken{
+			Identifier: token.Identifier,
+			Nonce:      token.Nonce,
+			Amount:     amountToDeposit,
+			Type:       token.Type,
+		}
+
+		// deposit a prefixed token from main chain to sovereign chain,
+		// expecting these tokens to be burned by the whitelisted ESDT safe sc
+		txResult = Deposit(t, cs, receiver.Bytes, &receiverNonce, bridgeData.ESDTSafeAddress, []chainSim.ArgsDepositToken{bridgedOutToken}, receiver.Bytes)
+		chainSim.RequireSuccessfulTransaction(t, txResult)
+
+		remainingTokenAmount := big.NewInt(0).Sub(receiverToken.Amount, amountToDeposit)
+		checkTokenDataInAccount(t, cs, token, receiver.Bech32, remainingTokenAmount)
+		checkTokenDataInAccount(t, cs, token, esdtSafeEncoded, big.NewInt(0))
+		receiverToken.Amount = remainingTokenAmount
+
+		if remainingTokenAmount.Cmp(big.NewInt(0)) > 0 {
```

Comment:
Why make this "same shard" transfer based on remaining amount?

You seem to always send tokens above from `receiver`, then in "same shard" you also send from `receiver`, so you should always have tokens in `receiver`

Reply:
- The `wallet` in shard 0 sends tokens to `receiver` in shard 2
- The `receiver` will deposit (burn) in the bridge contract in shard 1
- The `receiver` in shard 2 sends tokens to `receiver2` in same shard
This check is to skip the NFT tokens because after deposit quantity is 0

Reply:
Can we make the check based on type then?
E.g.: if the remaining amount is zero and the type is nft => correct behavior, otherwise fail, because this might hide possible bugs

Reply:
As we discusses in private, with the last commit this should be fixed :D